src/Voron/Debugging/StorageReportGenerator.cs

```diff
@@ -210,19 +211,20 @@ public static TreeReport GetReport(FixedSizeTree fst, bool calculateExactSizes)
                 PageCount = fst.PageCount,
                 Density = density,
                 AllocatedSpaceInBytes = fst.PageCount * Constants.Storage.PageSize,
-                UsedSpaceInBytes = calculateExactSizes ? (long)(fst.PageCount * Constants.Storage.PageSize * density) : -1,
-                MultiValues = null
+                UsedSpaceInBytes = includeDetails ? (long)(fst.PageCount * Constants.Storage.PageSize * density) : -1,
+                MultiValues = null,
             };
             return treeReport;
         }
 
-        public static TreeReport GetReport(Tree tree, bool calculateExactSizes)
+        public static TreeReport GetReport(Tree tree, bool includeDetails)
         {
             List<double> pageDensities = null;
-
-            if (calculateExactSizes)
+            Dictionary<int, int> pageBalance = null;
+            if (includeDetails)
             {
                 pageDensities = GetPageDensities(tree);
+                pageBalance = GatherBalanceDistribution(tree);
```

Comment:
I wonder if we could calculate both page densities and balance distribution at once and this way iterate over all pages of a tree one time

Reply:
No difference, the execution time is dominated hard by hard page faults. On codealike database it takes a minute the first run. Second is under a second. Also I was under the impression that the details enabled is not usually call.